### PR TITLE
PZ-862 delete cdk branch without underscores

### PR DESCRIPTION
**Pull request by maa37**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -59,9 +59,11 @@ runs:
       export UNESCAPED_BRANCH_NAME=${{ inputs.branch-name }}
       
       branch=$(echo "$UNESCAPED_BRANCH_NAME" | sed "s%refs\/heads\/%%g")
-      clean_branch=$(echo "$branch" | sed "s/_//g" | tr "[:upper:]" "[:lower:]" | cut -c -38)
+      cdk_branch=$(echo "$branch" | sed "s/_//g" | cut -c -38)
+      clean_branch=$(echo "$branch" tr "[:upper:]" "[:lower:]" )
```
Comment (joshfng) [inline action.yml]: I don't think the current `clean_branch` needs to change at all. Also I think this is producing unexpected values:

```shell
clean_branch=$(echo "$branch" tr "[:upper:]" "[:lower:]" )                                 
echo $clean_branch
test_it-out_123 tr [:upper:] [:lower:]
```

**Review by nateoneill4435 — DISMISSED** [review]

**Review by joshfng — COMMENTED** [review]
I think the `clean_branch` code should remain the same, as it's producing an incorrect result right now. See my comment on the diff.

I'm also not sure why we can't use the existing `$clean_branch` value for the cdk name. It looks to produce the same results

```shell
branch=test_it-out_123
echo $branch
test_it-out_123

# the current method
clean_branch=$(echo "$branch" | sed "s/_//g" | tr "[:upper:]" "[:lower:]" | cut -c -38)
echo $clean_branch 
testit-out123

# the changes after this diff which I believe are just a typo needing fixed
clean_branch=$(echo "$branch" tr "[:upper:]" "[:lower:]" )
echo $clean_branch 
test_it-out_123 tr [:upper:] [:lower:]

# the new cdk_branch
cdk_branch=$(echo "$branch" | sed "s/_//g" | cut -c -38)
echo $cdk_branch 
testit-out123
```

Both `clean_branch` and `cdk_branch` are producng the same string `test_it-out_123` becomes `testit-out123`

File: action.yml (hunk 1 above)
Comment (joshfng) [reply action.yml]: You're missing a pipe operator

```shell
clean_branch=$(echo "$branch" | tr "[:upper:]" "[:lower:]" )                               
echo $clean_branch                                          
test_it-out_123
```

But I think we should still have the `sed "s/_//g"` as well. In that case, the current stuff should stay the same and we should just pass `$clean_branch` to the cdk stack name

File: action.yml [hunk 2]
```diff
@@ -59,9 +59,11 @@ runs:
       export UNESCAPED_BRANCH_NAME=${{ inputs.branch-name }}
       
       branch=$(echo "$UNESCAPED_BRANCH_NAME" | sed "s%refs\/heads\/%%g")
-      clean_branch=$(echo "$branch" | sed "s/_//g" | tr "[:upper:]" "[:lower:]" | cut -c -38)
+      cdk_branch=$(echo "$branch" | sed "s/_//g" | cut -c -38)
+      clean_branch=$(echo "$cdk_branch" tr "[:upper:]" "[:lower:]" )
```
Comment (joshfng) [inline action.yml]:
```suggestion
      clean_branch=$(echo "$cdk_branch" | tr "[:upper:]" "[:lower:]" )
```
Forgot a pipe

**Review by joshfng — CHANGES_REQUESTED** [review]

**Review by joshfng — APPROVED** [review]

**Pull request merged**